docs/src/components/ComponentPlayground/ComponentPlaygroundSnippet.tsx

```diff
@@ -0,0 +1,35 @@
+import { CodeSnippet, CodeSnippetProps } from '@stardust-ui/docs-components'
+import * as React from 'react'
+
+import renderElementToJSX from 'docs/src/components/ExampleSnippet/renderElementToJSX'
+
+type ComponentPlaygroundSnippetProps = {
+  element?: React.ReactElement
+  component?: React.FunctionComponent
+} & Partial<CodeSnippetProps>
+
+/**
+ * This component uses `react-element-to-jsx-string` to get a generated markup with props.
+ */
+const ComponentPlaygroundSnippet: React.FunctionComponent<
+  ComponentPlaygroundSnippetProps
+> = props => {
+  const { element, component, ...rest } = props
+
+  if (process.env.NODE_ENV !== 'production') {
+    if (typeof component === 'function' && !!component.prototype.isReactComponent) {
```

Comment:
this is pretty ugly... React does not offer any helper methods for this? similar to `React.isValidElement`

Reply:
Nope, as I know because for React there is no difference: https://stackoverflow.com/a/52211951/6488546

Actually, this component is quite hacky and tricky 😒